cli/build.go

```diff
@@ -49,6 +49,12 @@ var BuildCommand = &cli.Command{
 		},
 	}, commonPlanFlags()...),
 	Action: func(ctx context.Context, cmd *cli.Command) error {
+		platform, err := getPlatform(cmd.String("platform"))
+		if err != nil {
+			return cli.Exit(err, 1)
+		}
+		os.Setenv("TARGETPLATFORM", platform.String())
```

Comment:
this is a bad idea(tm) going to think of an alternative solution to passing the platform through. we have to be aware of this in the plan afaict so I'll have to figure out how to get it into the build context I guess.